Admin/Admin.php

```diff
@@ -1,326 +1,368 @@
+                    ->save($entity);
+            }
+            // inform user everything went fine
+            $this
+                ->messageHandler
+                ->handleSuccess('lag.admin.' . $this->name . '.saved');
+            $success = true;
+        } catch (Exception $e) {
+            $this
+                ->messageHandler
+                ->handleError(
+                    'lag.admin.saved_errors',
+                    "An error has occurred while saving an entity : {$e->getMessage()}, stackTrace: {$e->getTraceAsString()} "
+                );
+            $success = false;
+        }
+        return $success;
     }
 
     /**
-     * @return mixed
+     * Delete entity via admin manager. Error are catch, logged and an flash message is added
+     *
+     * @return bool true if the entity was saved without errors
      */
-    public function getController()
+    public function delete()
     {
-        return $this->controller;
+        try {
+            foreach ($this->entities as $entity) {
+                $this
+                    ->manager
+                    ->delete($entity);
+            }
+            // inform user everything went fine
+            $this
+                ->messageHandler
+                ->handleSuccess('lag.admin.' . $this->name . '.deleted');
+            $success = true;
+        } catch (Exception $e) {
+            $this
+                ->messageHandler
+                ->handleError(
+                    'lag.admin.deleted_errors',
+                    "An error has occurred while deleting an entity : {$e->getMessage()}, stackTrace: {$e->getTraceAsString()} "
+                );
+            $success = false;
+        }
+        return $success;
     }
 
     /**
-     * Return entity for current admin. If entity does not exist, it throws an exception
+     * Generate a route for admin and action name (like lag.admin.my_admin)
+     *
+     * @param $actionName
+     *
+     * @return string
      *
-     * @return mixed
      * @throws Exception
      */
-    public function getEntity()
+    public function generateRouteName($actionName)
     {
-        if (!$this->entity) {
-            throw new Exception("Entity not found in admin \"{$this->getName()}\". Try call method findEntity or createEntity first.");
+        if (!array_key_exists($actionName, $this->getConfiguration()->getActions())) {
+            $message = 'Invalid action name %s for admin %s (available action are: %s)';
+            throw new Exception(sprintf($message, $actionName, $this->getName(), implode(', ', $this->getActionNames())));
         }
-        return $this->entity;
-    }
-
-    public function getEntityLabel()
-    {
-        $label = '';
+        // get routing name pattern
+        $routingPattern = $this->getConfiguration()->getRoutingNamePattern();
+        // replace admin and action name in pattern
+        $routeName = str_replace('{admin}', $this->underscore($this->getName()), $routingPattern);
```

Comment:
Yes we can use sprintf here but it will have interest. In this code, I just will replace one the admin token and action token, and admin token is optional. I think the code is more readable as is.